test/jdk/javax/swing/JRadioButton/bug4380543.java

```diff
@@ -0,0 +1,160 @@
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class bug4380543 {
+    static testFrame testObj;
+    static String instructions
+            = "INSTRUCTIONS:" +
+            "\n 1. This is a Windows specific test. If you are not on " +
+            "Windows, press Pass." +
+            "\n 2. Check if the Left insets(margins) is set visually " +
+            "similar to other three sides around Radio Button and CheckBox" +
+            "(insets set to 20 on all 4 sides)." +
+            "\n 3. If Left insets(margins) appear Empty, press Fail, " +
+            "else press Pass.";
+    static PassFailJFrame passFailJFrame;
+
+    public static void main(String args[]) throws Exception {
+
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    passFailJFrame = new PassFailJFrame(instructions);
+                    testObj = new testFrame();
+                    //Adding the Test Frame to handle dispose
+                    PassFailJFrame.addTestFrame(testObj);
+                    PassFailJFrame.positionTestFrame(testObj, PassFailJFrame.Position.HORIZONTAL);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+        passFailJFrame.awaitAndCheck();
+    }
+}
+
+class testFrame extends JFrame implements ActionListener {
+    JPanel buttonsPanel;
+
+    Map<String, String> lookAndFeelMaps = new HashMap<String, String>();
+    public testFrame() throws InterruptedException {
+        initMap();
+        initComponents();
+    }
+
+    public void initMap()
+    {
+        String sLnF;
+        String sMapKey;
+        UIManager.LookAndFeelInfo[] lookAndFeel = UIManager.getInstalledLookAndFeels();
+        for (UIManager.LookAndFeelInfo look : lookAndFeel) {
+
+            sLnF = look.getClassName();
+            sMapKey = sLnF.substring(sLnF.lastIndexOf(".")+1);
+            sMapKey = sMapKey.replaceAll("LookAndFeel","");
+            sMapKey = sMapKey.trim();
+
+            lookAndFeelMaps.put(sMapKey, sLnF);
+        }
+    }
+
+    public void initComponents() throws InterruptedException {
+        JPanel p = new JPanel();
+        buttonsPanel = new JPanel();
+        buttonsPanel.setLayout(new BoxLayout(buttonsPanel, BoxLayout.Y_AXIS));
+
+        JRadioButton rb  = new JRadioButton("JRadioButton");
+        rb.setMargin(new Insets(20, 20, 20, 20));
+        rb.setBackground(Color.green);
+        rb.setAlignmentX(0.5f);
+        buttonsPanel.add(rb);
+
+        JCheckBox cb  = new JCheckBox("JCheckBox");
+        cb.setMargin(new Insets(20, 20, 20, 20));
+        cb.setBackground(Color.yellow);
+        cb.setAlignmentX(0.5f);
+        buttonsPanel.add(cb);
+
+        getContentPane().add(buttonsPanel);
+
+        for (Map.Entry mapElement : lookAndFeelMaps.entrySet()) {
```

Comment:
```suggestion
        for (Map.Entry<String, String> mapElement : lookAndFeelMaps.entrySet()) {
```

Use parametrized `Map.Entry`.